cocos2d/core/renderer/RendererCanvas.js

```diff
@@ -37,24 +37,166 @@ cc.rendererCanvas = {
     _clearColor: cc.color(),                                  //background color,default BLACK
     _clearFillStyle: "rgb(0, 0, 0)",
 
+    _dirtyRegion: null,
+    _allNeedDraw: true,
+    _enableDirtyRegion: false,
+    _debugDirtyRegion: false,
+    _canUseDirtyRegion: false,
+    //max dirty Region count, default is 10
+    _dirtyRegionCountThreshold: 10,
+
     getRenderCmd: function (renderableObject) {
         //TODO Add renderCmd pool here
         return renderableObject._createRenderCmd();
     },
 
+    enableDirtyRegion: function (enabled) {
+        this._enableDirtyRegion = enabled;
+    },
+
+    isDirtyRegionEnabled: function () {
+        return this._enableDirtyRegion;
+    },
+
+    setDirtyRegionCountThreshold: function(threshold) {
+        this._dirtyRegionCountThreshold = threshold;
+    },
+
+    _collectDirtyRegion: function () {
```

Comment:
suggest to cache RegionStatus here and use the local variable in the following logic

```
var Status = cc.Node.CanvasRenderCmd.RegionStatus;
```